src/wallet/scriptpubkeyman.cpp

```diff
@@ -652,7 +652,7 @@ std::optional<PSBTError> LegacyScriptPubKeyMan::FillPSBT(PartiallySignedTransact
         // Check non_witness_utxo has specified prevout
         if (input.non_witness_utxo) {
             if (txin.prevout.n >= input.non_witness_utxo->vout.size()) {
-                return PSBTError::MISSING_INPUTS;
+                return PSBTError::INPUTS_MISSING_OR_SPENT;
```

Comment:
In commit "refactor: PSBTError::MISSING_INPUTS" (b2b4b932572c5bb1ffa3fc4f34e17130348fbc24)

Does the name INPUTS_MISSING_OR_SPENT actually make sense to describe these two errors from FillPSBT? I'm not sure, but it seems these errors would only happen if the PSBT data was not internally consistent, so maybe it would be more appropriate to call this error something like PSBTError::INPUTS_INVALID?

Reply:
Yes I agree on re-reading this that this error does not quite correctly describe the specific check being failed here, which is that the outpoints vout is not higher than the number of vouts in the input, i.e. it is by definition invalid.

I have taken your suggestion on the wording in 77ef75d3ce4bff76a0db730b57896dcf7e8f3988